lib/link_checker/uri_checker/http_checker.rb

```diff
@@ -119,13 +128,14 @@ def call
     INVALID_TOP_LEVEL_DOMAINS = %w(xxx adult dating porn sex sexy singles).freeze
     REDIRECT_STATUS_CODES = [301, 302, 303, 307, 308].freeze
     REDIRECT_LIMIT = 8
+    REDIRECT_LOOP_LIMIT = 1
     REDIRECT_WARNING = 2
     RESPONSE_TIME_LIMIT = 15
     RESPONSE_TIME_WARNING = 5
 
     def check_redirects
       add_problem(TooManyRedirects.new(from_redirect: from_redirect?)) if redirect_history.length >= REDIRECT_LIMIT
-      add_problem(RedirectLoop.new(from_redirect: from_redirect?)) if redirect_history.include?(uri)
+      add_problem(RedirectLoop.new(from_redirect: from_redirect?)) if redirect_history.count(uri) > REDIRECT_LOOP_LIMIT
```

Comment:
I think it would be better if this constant was set to 2 and followed the same `>=` behaviour of the line above, otherwise the limit values mean something slightly different.